src/journal/journal-send.c

```diff
@@ -73,27 +74,39 @@ _public_ int sd_journal_print(int priority, const char *format, ...) {
 }
 
 _public_ int sd_journal_printv(int priority, const char *format, va_list ap) {
-
-        /* FIXME: Instead of limiting things to LINE_MAX we could do a
-           C99 variable-length array on the stack here in a loop. */
-
-        char buffer[8 + LINE_MAX], p[STRLEN("PRIORITY=") + DECIMAL_STR_MAX(int) + 1];
+        char p[STRLEN("PRIORITY=") + DECIMAL_STR_MAX(int) + 1];
+        char sbuf[LINE_MAX] = "MESSAGE=";
         struct iovec iov[2];
+        unsigned long len;
```

Comment:
hmm, vsnprintf() returns "int", not "unsigned long"

Reply:
Fixed.